net/InterfaceController.c

```diff
@@ -1182,10 +1182,13 @@ struct InterfaceController* InterfaceController_new(struct CryptoAuth* ca,
             : NULL
 
     }), sizeof(struct InterfaceController_pvt));
-    Identity_set(out);
+    Bits_memcpy(out->beacon.publicKey, ca->publicKey, 32);
 
-    out->icis = ArrayList_OfIfaces_new(alloc);
+    // Timeout here to avoid race condition
+    Timeout_setTimeout(beaconInterval, out, BEACON_INTERVAL, eventBase, alloc);
```

Comment:
I understand the entire re-shuffle here is about that race condition, what is racing ?